.github/workflows/oas.yml

```diff
@@ -40,22 +39,19 @@ jobs:
         make
         pip install -e .
         cd ..
-        git clone https://github.com/mdolab/pygeo.git
-        pip install -e ./pygeo
+        pip install "pygeo[testing] @ git+https://github.com/mdolab/pygeo.git"
 
     - name: Run tests
       env:
         OMPI_MCA_btl: ^openib   # prevent OpenMPI warning messages
       run: |
-        cd openaerostruct
-        testflo -n 2 openaerostruct --coverage --coverpkg openaerostruct --cover-omit \*tests/\* --cover-omit \*docs/\*
+        python -m pytest --cov-config=.coveragerc --cov=openaerostruct --cov-report=xml
```

Comment:
Why the switch to `pytest` from `testflo`?

Reply:
There appears to be no support in testflo for coverage file formats compatible codecov (xml, json, or txt). We were seeing the same problems as I ran into in OpenConcept when I didn't use the --cov-report=xml flag. Maybe there's a solution with testflo given that it's used on other repos? Though you suggested that pytest is much better supported anyway, so we figured it couldn't hurt.

Reply:
We run `coverage` manually to convert the coverage report to xml, see [here](https://github.com/mdolab/.github/blob/master/azure/azure_build.yaml#L112) for the implementation on Azure. If you're not using any testflo features and you don't plan to (MPI testing primarily), then this is fine. I just find it a bit strange given `testflo` is developed by the OM team but no strong objections here.

Reply:
Yeah this is weird, I use testflo in my FEMpy repo and am able to upload to codecov without doing any conversion to xml, not saying you need to figure it out if what you have setup works, but it's definitely possible in general.

Reply:
You seem to be using v1 of the action, where they supported the other format but it's deprecated. They dropped it in v2 --- there have been complaints on their repo since this was an undocumented change.

Reply:
I'm going to commit most of the changes that were brought up shortly and make sure the tests pass using `pytest` after the fix to the random seeds.  If the tests pass, I can change the gh action to manually run `coverage` and revert back to `testflo` to stay consistent with the previous test framework if that seems like the best route.